packages/block/src/block.ts

```diff
@@ -49,6 +49,7 @@ export class Block {
     // parse uncle headers
     const uncleHeaders = []
     for (const uhData of uhsData ?? []) {
+      header._common.setHardforkByBlockNumber(uhData.number!)
```

Comment:
would it be possible to use the block option `hardforkByBlockNumber` in the `BlockHeader.fromHeaderData` below instead?

Reply:
Oh that's a great suggestion! However, we should then explicitly revert back into `block.ts` to the old common, since it edits the common directly (and does not clone it).

Reply:
Hmm, I guess I should then cache the current fork before creating the uncle header, right? Otherwise it is always assumed that we use `hardforkByBlockNumber`?

Reply:
Can we also have one or two tests in the block library itself on this change?

Reply:
Good point, I will write some tests, but at this point I think the implementation is not really consistent yet. I think it would make sense to check if block options has set `hardforkByBlockNumber` set explicitly to `false`, if that's the case then also don't set the fork on the uncles, because that's a bit inconsistent at this point.

Reply:
looks great!

Reply:
@jochem-brouwer ok, thanks for the additional context.